src/data/reports/glass-admin/GLASSDataMaintenanceDefaultRepository.ts

```diff
@@ -0,0 +1,492 @@
+    Paging,
+    Sorting,
+    getPaginatedObjects,
+} from "../../../domain/common/entities/PaginatedObjects";
+
+interface ATCJson {
+    [key: string]: any;
+}
+
+export class GLASSDataMaintenanceDefaultRepository implements GLASSDataMaintenanceRepository {
+    private storageClient: StorageClient;
+    private globalStorageClient: StorageClient;
+
+    constructor(private api: D2Api) {
+        const instance = new Instance({ url: this.api.baseUrl });
+        this.storageClient = new DataStoreStorageClient("user", instance);
+        this.globalStorageClient = new DataStoreStorageClient("global", instance);
+    }
+
+    async get(
+        options: GLASSDataMaintenanceOptions,
+        namespace: string
+    ): Promise<GLASSMaintenancePaginatedObjects<GLASSDataMaintenanceItem>> {
+        const { paging, sorting, module } = options;
+        if (!module) return emptyPage;
+
+        const uploads = await this.getUploads(namespace);
+        const countries = await this.getCountries();
+
+        const filteredFiles = this.getFilteredFiles(uploads, countries, module);
+
+        const rowIds = this.getRowIds(filteredFiles);
+        const { objects, pager } = this.paginate(filteredFiles, sorting, paging);
+
+        return { objects: objects, pager: pager, rowIds: rowIds };
+    }
+
+    async getATCs(options: ATCOptions, namespace: string): Promise<PaginatedObjects<ATCItem>> {
+        const { paging, sorting } = options;
+
+        const atcs = await this.getATCItems(namespace);
+        const { objects, pager } = this.paginate(atcs, sorting, paging);
+
+        return { objects: objects, pager: pager };
+    }
+
+    private async getATCItems(namespace: string) {
+        return (await this.globalStorageClient.getObject<ATCItem[]>(namespace)) ?? [];
+    }
+
+    async getUserModules(config: Config): Promise<GLASSModule[]> {
+        const modules = await this.getModules();
+
+        return getUserModules(modules, config.currentUser);
+    }
+
+    async delete(namespace: string, items: Id[]): Promise<void> {
+        await this.deleteFileResources(items);
+
+        const uploads = await this.getUploads(namespace);
+        const updatedUploads = this.updateStatusById(items, uploads, "DELETED");
+        return await this.globalStorageClient.saveObject<GLASSDataMaintenanceItem[]>(namespace, updatedUploads);
+    }
+
+    async getColumns(namespace: string): Promise<string[]> {
+        const columns = await this.storageClient.getObject<string[]>(namespace);
+
+        return columns ?? [];
+    }
+
+    async saveColumns(namespace: string, columns: string[]): Promise<void> {
+        return this.storageClient.saveObject<string[]>(namespace, columns);
+    }
+
+    async uploadATC(namespace: string, file: File, year: string, items: ATCItemIdentifier[]): Promise<void> {
+        const atcItems = await this.getATCItems(namespace);
+        const jsons = await this.extractJsonFromZIP(file);
+
+        if (jsons.length === 4) {
```

Comment:
Now we change the number of jsons to **6**. `ddd_alterations` and `atc_alterations` jsons have been added together with `ddd`, `atc`, `ddd_combinations` and `conversion`